core/workflow-core/src/main/scala/edu/uci/ics/amber/core/storage/result/iceberg/IcebergTableManager.scala

```diff
@@ -0,0 +1,213 @@
+    }
+
+    // Convert mutable stats to immutable map
+    stats.map {
+      case (field, fieldStats) =>
+        field -> Map(
+          "min" -> fieldStats("min"),
+          "max" -> fieldStats("max"),
+          "mean" -> (if (fieldStats("count") > 0) fieldStats("sum") / fieldStats("count")
+                     else Double.NaN)
+        )
+    }.toMap
+  }
+
+  /**
+    * Calculates statistics for date fields in the table.
+    * The statistics include minimum and maximum dates for each field.
+    *
+    * @return A map where each key is a date field name, and the value is a map of statistics.
+    */
+  def calculateDateStats(): Map[String, Map[String, Date]] = {
+    val dateFields = detectDateFields()
+
+    val stats = mutable.Map[String, mutable.Map[String, Date]]()
+
+    dateFields.foreach { field =>
+      stats(field) = mutable.Map("min" -> new Date(Long.MaxValue), "max" -> new Date(Long.MinValue))
+    }
+
+    sampleDocuments().foreach { record =>
+      dateFields.foreach { field =>
+        val value = Option(record.getField(field)).map {
+          case ldt: java.time.LocalDateTime =>
+            Date.from(ldt.atZone(ZoneId.systemDefault()).toInstant)
+          case other =>
+            throw new IllegalArgumentException(s"Unexpected type for field '$field': $other")
+        }
+        if (value.isDefined) {
+          val fieldStats = stats(field)
+          fieldStats("min") =
+            if (value.get.before(fieldStats("min"))) value.get else fieldStats("min")
+          fieldStats("max") =
+            if (value.get.after(fieldStats("max"))) value.get else fieldStats("max")
+        }
+      }
+    }
+
+    stats.map {
+      case (field, fieldStats) =>
+        field -> Map(
+          "min" -> fieldStats("min"),
+          "max" -> fieldStats("max")
+        )
+    }.toMap
+  }
+
+  /**
+    * Calculates statistics for categorical fields in the table.
+    * The statistics include counts of unique values for each field.
+    *
+    * @return A map where each key is a categorical field name, and the value is a map of counts.
+    */
+  def calculateCategoricalStats(): Map[String, Map[String, Map[String, Integer]]] = {
+    val categoricalFields = detectCategoricalFields()
+
+    val stats = mutable.Map[String, mutable.Map[String, Int]]()
+
+    categoricalFields.foreach { field =>
+      stats(field) = mutable.Map[String, Int]()
+    }
+
+    sampleDocuments().foreach { record =>
+      categoricalFields.foreach { field =>
+        val value = Option(record.getField(field)).map(_.toString).getOrElse("null")
+        stats(field).updateWith(value) {
+          case Some(count) => Some(count + 1)
+          case None        => Some(1)
+        }
+      }
+    }
+
+    stats.map {
+      case (field, fieldStats) =>
+        field -> Map(
+          "counts" -> fieldStats.toMap.map { case (key, value) => key -> Integer.valueOf(value) }
+        )
+    }.toMap
+  }
+
+  private def sampleDocuments(): Seq[Record] = {
```

Comment:
instead of doing sampling, can you try to use the apis like:
`file.file().valueCounts()`, `file.file().lowerBound()`, `file.file().upperBound()`

This can avoid opening the data file and sampling.